pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -88,6 +88,17 @@ def __init__(
     def root_device(self):
         return self.parallel_devices[self.local_rank]
 
+    @property
+    def num_nodes(self) -> int:
+        return self._num_nodes
+
+    @num_nodes.setter
+    def num_nodes(self, num_nodes: int) -> None:
+        # note that world ranks is related to num_nodes, when resetting these parameters,
+        # need to reset world ranks
+        self._num_nodes = num_nodes
+        self.set_world_ranks()
```

Comment:
should we mark the `num_nodes` and `sync_batchnorm` constructor args as deprecated? or at least add a warning that they will be overridden by the trainer based on the trainer settings?

Reply:
then I think we could make these args as Optional for now, and add warning that if passed explicitly.